integration/faucet/faucet_test.go

```diff
@@ -0,0 +1,113 @@
+//go:build !relayer
+// +build !relayer
+
+package faucet_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/require"
+	envtest "github.com/tendermint/starport/integration"
+	"github.com/tendermint/starport/starport/pkg/cosmosclient"
+	"github.com/tendermint/starport/starport/pkg/cosmosfaucet"
+	"github.com/tendermint/starport/starport/pkg/xurl"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	addr = "cosmos1zqr2gd7hwkyw55knad0l6ml6ngutd70878evqj"
+)
+
+var (
+	defaultCoins = []string{"10token", "1stake"}
+	maxCoins     = []string{"102token", "100000000stake"}
+)
+
+func TestRequestCoinsFromFaucet(t *testing.T) {
+	var (
+		env          = envtest.New(t)
+		apath        = env.Scaffold("faucet")
+		servers      = env.RandomizeServerPorts(apath, "")
+		faucetURL    = env.ConfigureFaucet(apath, "", defaultCoins, maxCoins)
+		ctx, cancel  = context.WithTimeout(env.Ctx(), envtest.ServeTimeout)
+		faucetClient = cosmosfaucet.NewClient(faucetURL)
+	)
+	isErrTransferRequest := func(err error, expectedCode int) {
+		require.ErrorAs(t, err, &cosmosfaucet.ErrTransferRequest{})
+		errTransfer := err.(cosmosfaucet.ErrTransferRequest)
+		require.EqualValues(t, expectedCode, errTransfer.StatusCode)
+	}
+
+	// serve the app
+	go func() {
+		env.Must(env.Serve("should serve app", apath, "", "", envtest.ExecCtx(ctx)))
+	}()
+
+	// wait servers to be online
+	defer cancel()
+	err := env.IsAppServed(ctx, servers)
+	require.NoError(t, err)
+
+	err = env.IsFaucetServed(ctx, faucetClient)
+	require.NoError(t, err)
+
+	// error "account doesn't have any balances" occurs if a sleep is not included
+	time.Sleep(time.Second * 1)
```

Comment:
It seems to me to be like replacing a hack with another hack and it makes the test more complicated since upon the first successful Transfer request where the error is asserted, the balance will be different

The actual fix IMO should be that `FaucetInfo` endpoint successful returns when the account is operational or having `FaucetStatus`

I don't think this sleep is a big issue if it's outside of the relayer logic